index.html

```diff
@@ -1836,26 +1836,77 @@ <h2>
 
     <p>
       The {{MulticastController}} interface provides methods to join and leave multicast groups and to inspect
-      which groups have been joined. An instance of this interface is obtained from the
-      {{UDPSocketOpenInfo/multicastController}} member of the dictionary returned by the {{UDPSocket/opened}} promise.
+      which groups have been joined. It supports both Any-Source Multicast (ASM) and Source-Specific Multicast (SSM).
+      An instance of this interface is obtained from the {{UDPSocketOpenInfo/multicastController}} member of the
+      dictionary returned by the {{UDPSocket/opened}} promise.
       This member is only present if the socket is in {{UDPSocket/bound}} {{UDPSocket/mode}} and
       and [=this=]'s [=relevant global object=]'s [=associated Document=] is [=allowed to use=]
       the [=policy-controlled feature=] named "[=policy-controlled feature/direct-sockets-multicast=]".
     </p>
 
     <pre class="idl">
+      dictionary MulticastGroupOptions {
+        DOMString sourceAddress;
+      };
+
+      dictionary MulticastMembership {
+        required DOMString groupAddress;
```

Comment:
The `required` attribute doesn't make a difference for dictionaries returned by APIs since the behavior is defined by the algorithmic steps and not the WebIDL.

```suggestion
        DOMString groupAddress;
```